src/Sylius/Component/Core/spec/Promotion/Action/ItemFixedDiscountActionSpec.php

```diff
@@ -0,0 +1,240 @@
+        $originator,
+        $priceRangeFilter,
+        $taxonFilter,
+        AdjustmentInterface $promotionAdjustment1,
+        AdjustmentInterface $promotionAdjustment2,
+        Collection $originalItems,
+        Collection $units,
+        OrderInterface $order,
+        OrderItemInterface $orderItem1,
+        OrderItemInterface $orderItem2,
+        OrderItemInterface $orderItem3,
+        OrderItemUnitInterface $unit1,
+        OrderItemUnitInterface $unit2,
+        PromotionInterface $promotion
+    ) {
+        $order->getItems()->willReturn($originalItems);
+        $originalItems->toArray()->willReturn([$orderItem1, $orderItem2, $orderItem3]);
+
+        $priceRangeFilter
+            ->filter([$orderItem1, $orderItem2, $orderItem3], ['amount' => 500, 'filters' => ['taxons' => ['testTaxon']]])
+            ->willReturn([$orderItem1, $orderItem2])
+        ;
+        $taxonFilter
+            ->filter([$orderItem1, $orderItem2], ['amount' => 500, 'filters' => ['taxons' => ['testTaxon']]])
+            ->willReturn([$orderItem1])
+        ;
+
+        $orderItem1->getQuantity()->willReturn(2);
+        $orderItem1->getUnits()->willReturn($units);
+        $units->getIterator()->willReturn(new \ArrayIterator([$unit1->getWrappedObject(), $unit2->getWrappedObject()]));
+
+        $promotion->getDescription()->willReturn('Test description');
+
+        $adjustmentFactory->createNew()->willReturn($promotionAdjustment1, $promotionAdjustment2);
+
+        $unit1->getTotal()->willReturn(1000);
+        $promotionAdjustment1->setType(AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT)->shouldBeCalled();
+        $promotionAdjustment1->setLabel('Test description')->shouldBeCalled();
+        $promotionAdjustment1->setAmount(-500)->shouldBeCalled();
+
+        $originator->setOrigin($promotionAdjustment1, $promotion)->shouldBeCalled();
+
+        $unit2->getTotal()->willReturn(1000);
+        $promotionAdjustment2->setType(AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT)->shouldBeCalled();
+        $promotionAdjustment2->setLabel('Test description')->shouldBeCalled();
+        $promotionAdjustment2->setAmount(-500)->shouldBeCalled();
+
+        $originator->setOrigin($promotionAdjustment2, $promotion)->shouldBeCalled();
+
+        $unit1->addAdjustment($promotionAdjustment1)->shouldBeCalled();
+        $unit2->addAdjustment($promotionAdjustment2)->shouldBeCalled();
+
+        $this->execute($order, ['amount' => 500, 'filters' => ['taxons' => ['testTaxon']]], $promotion);
+    }
+
+    function it_does_not_apply_promotions_with_amount_0(
+        $adjustmentFactory,
+        OrderInterface $order,
+        OrderItemUnitInterface $unit1,
+        OrderItemUnitInterface $unit2,
+        PromotionInterface $promotion
+    ) {
+        $adjustmentFactory->createNew()->shouldNotBeCalled();;
+
+        $unit1->addAdjustment(Argument::any())->shouldNotBeCalled();
+        $unit2->addAdjustment(Argument::any())->shouldNotBeCalled();
```

Comment:
These two checks don't make any sense since we do not pass these objects to a method and no collaborator returns them.